configure.py

```diff
@@ -1753,15 +1753,11 @@ def configure_seastar(build_dir, mode, mode_config):
 
 
 def configure_abseil(build_dir, mode, mode_config):
-    # for sanitizer cflags
-    seastar_flags = query_seastar_flags(f'{outdir}/{mode}/seastar/seastar.pc',
-                                        mode_config['build_seastar_shared_libs'],
-                                        args.staticcxx)
-    seastar_cflags = seastar_flags['seastar_cflags']
+    abseil_cflags = mode_config['lib_cflags']
+    if '-DSANITIZE' in mode_config['cxxflags']:
+        abseil_cflags += ' -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr'
+    abseil_cflags = abseil_cflags.strip()
```

Comment:
Doesn't that mean that abseil's translation units will be built with different flags that scylladb's? And isn't it problematic since both include abseil's headers?

Reply:
that's correct. but almost all 3rd party libraries packages by distro maintainers are compiled with different flags than scylladb's. but we survived since the day 0. what do you mean by "both"?

Reply:
both scylladb/service/something.cc and scylladb/abseil/another/thing/blah.cpp include scylladb/abseil/another/thing/blah.h. An inline  function that is materialized in both translation units can behave differently.

We already saw some problems with -march=native and dpdk. I don't expect similar problems with abseil, but isn't mixing compiler option sets asking for trouble?

Reply:
> both scylladb/service/something.cc and scylladb/abseil/another/thing/blah.cpp include scylladb/abseil/another/thing/blah.h. An inline function that is materialized in both translation units can behave differently.

indeed, the same applies to the builds before we ran into #18507. but yeah, i agree with you, because of #18507 . 

> 
> We already saw some problems with -march=native and dpdk. I don't expect similar problems with abseil, but isn't mixing compiler option sets asking for trouble?

if you are referring https://github.com/scylladb/seastar/issues/2242#issuecomment-2113905363 . i explained in the comment. that was a bug in Clang.

again, i agree with you. but it depends.

Reply:
checked and created #19160